packages/settingeditor/src/settingspanel.tsx

```diff
@@ -107,7 +141,9 @@ export const SettingsPanel: React.FC<ISettingsPanelProps> = ({
 
   return (
     <div className="jp-SettingsPanel" ref={wrapperRef}>
-      {settings.map(pluginSettings => {
+      {shownPlugins.map(pluginSettings => {
+        // Pass filtered results to SettingsFormEditor to only display filtered fields.
+        const filtered = filterPlugin(pluginSettings.plugin);
```

Comment:
This is suboptimal because you are requesting again the plugins although the filter that produces `shownPlugins` has already been called earlier.

You should change the logic to store the filtered fields per plugin map as state. So you don't need to apply the filter except when `updateFilterSignal` is emitted.